typing/typeclass.ml

```diff
@@ -259,54 +259,56 @@ let make_method loc cl_num expr =
 let delayed_meth_specs = ref []
 
 let rec class_type_field env sign self_scope ctf =
-  Builtin_attributes.warning_scope ctf.pctf_attributes
-    (fun () -> class_type_field_aux env sign self_scope ctf)
-
-and class_type_field_aux env sign self_scope ctf =
   let loc = ctf.pctf_loc in
   let mkctf desc =
     { ctf_desc = desc; ctf_loc = loc; ctf_attributes = ctf.pctf_attributes }
   in
   match ctf.pctf_desc with
   | Pctf_inherit sparent ->
-      let parent = class_type env Virtual self_scope sparent in
-      complete_class_type parent.cltyp_loc
-        env Virtual Class_type parent.cltyp_type;
-      inherit_class_type ~strict:false loc env sign parent.cltyp_type;
-      mkctf (Tctf_inherit parent)
-
+      Builtin_attributes.warning_scope ctf.pctf_attributes
```

Comment:
It's a bit unfortunate (albeit very minor) that we need to duplicate the "logic" to handle pctf_attributes for all cases (except Pctf_attribute).  One could have another helper function that encapsule the use of `Builtin_attributes.warning_scope ctf.pctf_attributes` and `mkctf`:

````ocaml
let mkctf_attr f = Builtin_attributes.warning_scope ctf.pctf_attributes (mkctf (f ())) in
````

Reply:
That does seem slightly better.

Reply:
Fixed